projects/jdwp/runtime/async_streams.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+import asyncio
+import abc
+
+
+class JDWPInputStreamBase(abc.ABC):
+    def __init__(self, reader: asyncio.StreamReader) -> None:
+        self.reader = reader
+
```

Comment:
could you remove constructors ? There is no need to enforce reader attribute here